lib/hmr.js

```diff
@@ -0,0 +1,23 @@
+const socket = new WebSocket('ws://127.0.0.1:7778');
+
+
+function handleChange(file) {
+  const ext = file.split('.').pop();
+
+  if (ext === 'css') {
+    const link = document.querySelector(`link[rel="stylesheet"][href*="${file}"]`);
+
+    if (link) {
+      const href = link.href;
+      link.href = href;
+    }
+  } else if (['js', 'beard'].includes(ext)) {
+    // reload on frontend js, handles, and template changes
+    location.reload();
+  }
+}
+
+
+socket.onmessage = (e) => {
+  JSON.parse(e.data).forEach(handleChange);
```

Comment:
I think we already discussed this, so forgive me if I'm bringing up something we already discussed, but shouldn't this be something more like:

```
socket.onmessage = (e) => {
  handleChanges(JSON.parse(e.data));
}

function handleChanges(files) {
  const exts = files.map((f) => f.split('.').pop();

  if (exts.filter(ext => ['js', 'beard'].includes(ext)).length) {
    location.reload();
  } else {
    files.forEach(handleChange);
  }
}
```

Reply:
You might be misreading the `handleChange` function. That handles a single item from the changes that come through. Notice handleChange is the callback for the forEach so it's doing what you're conceptually advocating but just differently.

Reply:
Right, that's why I wrote a `handleChanges` function that handles all of them, and calls `handleChange` to handle the individual one.

It seems like we just need to call `location.reload()` if any single file has a js/beard extension, instead of iterating over all of them. However, I thought we already discussed this, and I'm sure you thought of this, so there may in fact be some reality of the existing code that I'm missing.